src/activities/quizzes/timing/QuizTimingEntity.js

```diff
@@ -79,13 +85,58 @@ export class QuizTimingEntity extends Entity {
 		return entity.properties && entity.properties.timeLimit;
 	}
 
-	async setTimingType(data) {
+	async setExtendedDeadline(data) {
 		if (!this.canEditTiming()) return;
+		const entity = this.getEnforcedTimingSubEntity();
+		if (!entity) return;
+		const subEntity = entity.getSubEntityByClass('automatic-zero');
```

Comment:
i think getting the `automatic-zero` entity can be a helper fn, since it should be used by the `getExtendedDeadline` function as well to get its value

Reply:
good idea, I made the change